### PR TITLE
Update parity-util-mem

### PR DESCRIPTION
**Pull request by cecton**
This will remove one extra dependency in substrate

```
[0] [11:01:30] ~/r/substrate cecton-cargo-update > cargo update -p parity-util-mem
error: There are multiple `parity-util-mem` packages in your project, and the specification `parity-util-mem` is ambiguous.
Please re-run this command with `-p <spec>` where `<spec>` is one of the following:
  parity-util-mem:0.3.0
  parity-util-mem:0.4.1
```

### CODE REVIEW

File: memory-db/Cargo.toml [hunk 1]
```diff
@@ -9,7 +9,7 @@ edition = "2018"
 
 [dependencies]
 heapsize = { version = "0.4", optional = true }
-parity-util-mem = { version = "0.3", default-features = false }
+parity-util-mem = { version = "0.4", default-features = false }
```
Comment (ordian) [inline memory-db/Cargo.toml]: the latest version is `0.5.1`, also this is a breaking change, do you want to submit a version bump in a separate PR? https://github.com/paritytech/trie/blob/master/CONTRIBUTING.md#releasing-a-new-version

Comment (cecton) [reply memory-db/Cargo.toml]: Is it a breaking change? parity-util-mem is not a direct dependency of memory-db. In fact, you could remove the dependency and it would work. But we need it to set some flag.

Comment (ordian) [reply memory-db/Cargo.toml]: > Is it a breaking change? parity-util-mem is not a direct dependency of memory-db. In fact, you could remove the dependency and it would work. But we need it to set some flag.

Not sure what you mean, it is a direct dependency (impls `parity_util_mem::MallocSizeOf` trait)
https://github.com/paritytech/trie/blob/a43b9a19e896533bd1bf6ca1ca12a81ab9b2c455/memory-db/src/lib.rs#L424-L463

If you remove it that would be removing a trait impl, which would break downstream crates that use this feature.

Comment (cecton) [reply memory-db/Cargo.toml]: Oh right!! Sorry I badly grepped the code (I forgot to change the `-` to `_`).

I will update to version 5 and bump the version.

Comment (cecton) [reply memory-db/Cargo.toml]: Good to merge then

**Comment by cecton** [comment]
```
[0] [13:44:09] ~/r/substrate cecton-cargo-update > cargo update
    Updating crates.io index
    Removing malloc_size_of_derive v0.1.1
    Removing memory-db v0.18.1
    Removing parity-util-mem v0.3.0
```

:rofl: this was totally not worth it
